framework/db/BaseActiveRecord.php

```diff
@@ -458,6 +458,12 @@ protected function createRelationQuery($class, $link, $multiple)
      */
     public function populateRelation($name, $records)
     {
+        foreach ($this->_relationsDependencies as $attribute => $relationNames) {
+            if (isset($relationNames[$name])) {
+                unset($this->_relationsDependencies[$attribute][$name]);
+            }
+        }
```

Comment:
doesn't this revert the whole idea of what `_relationsDependencies` is doing?

Reply:
the tests in https://github.com/yiisoft/yii2/pull/13618/files#diff-5c2b59d39a2ea1dc5443fb98d1325d11 do not cover eager loading. Pretty sure this gets broken by this change.

Reply:
Eager loading can have bug. Needs to be tested.

Reply:
relationsDependencies logic, do not works with eager loading initially, my pr do not change that behavior.